src/js/media-controller.js

```diff
@@ -532,16 +534,58 @@ const propagateMediaState = (els, stateName, val) => {
   });
 };
 
+const isPromiseLike = value => typeof value?.then === 'function';
+const isHTMLElement = value => value instanceof window.HTMLElement;
+
 const monitorMediaUIElementDescendantsOf = (root, associateCallback, unassociateCallback) => {
+
+  const associateCallbackWithAsync = (elOrPromise) => {
+    // Use structural type checking for "`then`" based on A+/Promise spec in case of 3rd party Promise impls or similar
+    // But also make sure it isn't an `HTMLElement`, just in case it happens to have a `then` method
+    if (isPromiseLike(elOrPromise) && !isHTMLElement(elOrPromise)) {
+      // Assume promises will yield the element when resolved
+      elOrPromise.then(el => {
+        const addedNodes = getMediaUIElementDescendants(el);
+        // Still use `associateCallbackWithAsync`, just in case more descendants aren't quite ready
+        addedNodes.forEach(associateCallbackWithAsync);
+      });
+      return;
+    }
+
+    associateCallback(elOrPromise);
+  };
+
+  const addedNodes = getMediaUIElementDescendants(root);
+  addedNodes.forEach(associateCallbackWithAsync);
+
+  /** @TODO Discuss recursively (un)associating or not, recursively monitoring or not, etc. for event use case (CJP) */
+  const associateElementHandler = (evt) => {
+    const el = evt?.composedPath()[0] ?? evt.target;
+    associateCallback(el);
+  };
+
+  const unassociateElementHandler = (evt) => {
+    const el = evt?.composedPath()[0] ?? evt.target;
+    unassociateCallback(el);
+  };
+
+  root.addEventListener(MediaUIEvents.MEDIA_ASSOCIATE_ELEMENT_REQUEST, associateElementHandler);
```

Comment:
Why does every root el need this handler?

Reply:
This was for consistency of the "escape hatch event". If someone is relying on this for a (likely dynamic sub-)component, this ensures that the event would work whether it's a direct descendant of `<media-controller/>` or is used in a "standalone component" manner.

Reply:
Got it. Makes sense.